### PR TITLE
Kitty terminal support

### PR DESCRIPTION
**Pull request by craigmac**
If this is accepted, I'll add to the same wiki page that shows wezterm
instructions on how you need to setup kitty for this to work.

Python files in are required to be in ~/.config/kitty/ folder to be found by
`kitty @ kitten ...` calls. Additionally, `~/.config/kitty/kitty.conf` must contain:

on macOS

```conf
allow_remote_control yes
listen_on unix:/tmp/mykitty
```

or if using Linux:

```conf
allow_remote_control yes
listen_on unix:@mykitty
```

and the bindings of course, in `~/.config/kitty/kitty.conf`

```conf
macos_option_as_alt both
map alt+j kitten pass_keys.py neighboring_window bottom alt+j
map alt+k kitten pass_keys.py neighboring_window top    alt+k
map alt+h kitten pass_keys.py neighboring_window left   alt+h
map alt+l kitten pass_keys.py neighboring_window right  alt+l
```

`macos_option_as_alt` is only required if you want to bind any
`alt+` on macOS kitty.

### CODE REVIEW

**Comment by craigmac** [comment]
closes #21

**Comment by craigmac** [comment]
should mention I haven't implemented the 'previous' binding yet, that one may be more difficult.

**Comment by numToStr** [comment]
Thanks for working on this. One thing that I am concerned about is the inclusion of python files inside the plugin. These are anyway required to be copied into kitty's config directory, so how about moving these files to wiki and giving proper instruction on how to use them?

Also, I am shocked to see that we need to do these kinds of shenanigans just to move b/w kitty panes. 🤷🏼

**Comment by craigmac** [comment]
Will do! I figured they're would be some issues with shipping python files, and having to do some post-install dance to get them in the right place (which would end up causing you a bunch of support noise). Yeah, this experience made me switch to wezterm over kitty permanently, just so I could have these sorts of things done directly in my lua config file.

**Comment by craigmac** [comment]
@numToStr done. I'll keep an eye on issues and help out if there's any issues with it.

File: lua/Navigator/mux/kitty.lua
Comment (numToStr) [inline lua/Navigator/mux/kitty.lua]:
```suggestion
    self.execute(string.format('kitten navigator.py %s', self.direction[direction]))
```

How about changing the filename?

**Review by numToStr — CHANGES_REQUESTED** [review]

File: lua/Navigator/mux/kitty.lua
Comment (craigmac) [reply lua/Navigator/mux/kitty.lua]: Done

**Comment by craigmac** [comment]
@numToStr anything else you'd like done?

**Comment by Arelav** [comment]
Hi. Thanks for an awesome plugin.
Does Kitty support finalized? I switched from Kitty to Wezterm a while ago and migrated to this plugin. It would be nice if I'll have both Kitty and Wezterm configured.
For now, I can't make it work in Kitty and I see that part of `mux/kitty.lua` wasn't merged yet.
Please confirm if it's not supported yet because I did everything according to Wiki page https://github.com/numToStr/Navigator.nvim/wiki/Kitty-Integration and can't make it work
@numToStr please approve this PR if it's ready to go
UPDATE: just to confirm, using the fork solves the issue and it works both in WezTerm and Kitty

**Comment by craigmac** [comment]
@Arelav the kitty instructions I've added to the wiki only work with the changes in this PR. Maybe I should either make that clear on the wiki page until this PR is merged. What do you think @numToStr? Anything left that you want to see before merging this?

**Comment by Arelav** [comment]
> @Arelav the kitty instructions I've added to the wiki only work with the changes in this PR. Maybe I should either make that clear on the wiki page until this PR is merged.

It's sad to say but looks like it won't be merged any soon. I still use your fork and it works just fine.

**Comment by craigmac** [comment]
@numToStr if you're not going to maintain this any longer, could you add me as maintainer? in the meantime if others are looking for the kitty support, it's on my fork, which I use and plan to maintain and improve.

**Comment by valerykh** [comment]
@craigmac as I understand you are maintainer now. Nice to see it's won't be abandoned anymore. Please let us know when it's good time to switch to main package. Or maybe we should continue using your fork. I'm using fork for a long time and no problems so far. Thank you for your work both. @numToStr @craigmac

**Comment by Arelav** [comment]
@craigmac as I understand you are maintainer now. Nice to see it's won't be abandoned anymore. Please let us know when it's good time to switch to main package. Or maybe we should continue using your fork. I'm using fork for a long time and no problems so far. Thank you for your work both. @numToStr @craigmac

**Comment by valerykh** [comment]
Or maybe I got this wrong. Looks like the PR was just decided by the author. So I'll stick to continue using fork then.

**Comment by Arelav** [comment]
Or maybe I got this wrong. Looks like the PR was just decided by the author. So I'll stick to continue using fork then.

**Pull request closed without merging**
